tracing-subscriber/src/registry/sharded.rs

```diff
@@ -176,11 +174,7 @@ impl<'a> LookupSpan<'a> for Registry {
 
 impl LookupMetadata for Registry {
     fn metadata(&self, id: &Id) -> Option<&'static Metadata<'static>> {
-        if let Some(span) = self.get(id) {
-            Some(span.metadata())
-        } else {
-            None
-        }
+        self.get(id).map(|data| data.metadata())
```

Comment:
TIOLI: could be `map(SpanData::metadata)`

Reply:
can't; the receiver isn't a `SpanData`, it's a `Guard<SpanData`.